### PR TITLE
XWIKI-22986: Username is displayed too small on Registration confirmation page and proper punctuation should be added

### PR DESCRIPTION
**Pull request by Sereza5**
# Jira URL

https://jira.xwiki.org/browse/XWIKI-22986

# Changes

## Description

* Fixed punctuation by updating the translation value and DOM representation.

## Clarifications

* Technically the `small` element[ shouldn't be used to show that some content has less importance](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-small-element) (second note). However it kinda makes sense here and it's convenient with our current design system. The semantics on this message are not that strong. As discussed in this ticket definition, it would also make complete sense to not include this distinction in our visuals.

# Screenshots & Video

Here are a couple screenshots of the changes:
Screenshot from 2025-03-27 15-00-35
Screenshot from 2025-03-27 14-59-53

# Executed Tests

Manual only, see screenshots above. No automated tests cover this UI.

# Expected merging strategy

* Prefers squash: Yes 
* Backport on branches:
  * 16.10.X (like the cause, [XWIKI-22121](https://jira.xwiki.org/browse/XWIKI-22121))

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties [hunk 1]
```diff
@@ -1686,7 +1686,7 @@ core.register.invalidUsername=Invalid username provided. Please use only letters
 core.register.mailSenderWronglyConfigured=The user has been created but the validation email has not been sent. Please check the Mail Sending Configuration and consider recreating the user.
 core.register.invalidCaptcha=Incorrect CAPTCHA answer.
 core.register.registerFailed=Registration has failed due to unknown reasons. (Error code: {0})
-core.register.successful.welcome=Welcome {0}
+core.register.successful.welcome=Welcome {0} {1}!
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties]: @surli after the issue last time when updating a translation key.
Can you confirm that it's indeed unsafe to perform this change and that it's better to introduce a new key?

Comment (michitux) [reply xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties]: It's clearly against our best practices at least: https://dev.xwiki.org/xwiki/bin/view/Community/L10N/Conventions/#HUpdatingtranslations (emphasis by me):

> It is a bad practice to do a major change on an existing message, such as changing the meaning or **adding parameters**. This means that existing translations in other languages will continue to use the old meaning, and this might "break" the application. It is better to start from scratch and create a different key.

Comment (surli) [reply xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties]: Yes clearly for such case a new translation key is needed.

Comment (Sereza5) [reply xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties]: Addressed this comment in https://github.com/xwiki/xwiki-platform/pull/4014/commits/70c32622820d201e7034ffcad2698d8437ede9ff :+1: 

I deprecated the old translation and created a new key with the second parameter.

File: xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml [hunk 2]
```diff
@@ -528,19 +528,22 @@
...
 #set($userLink = $xwiki.getUserName("$userSpace$userName"))
 #set($successAndLogin = $services.localization.render('core.register.successful.successandlogin'))
 [[image:registration_success_hero.svg||data-xwiki-image-style-alignment="center" height="50vh"]]
+#define($headlineSmall)
+#if(!$noReadableName)
+&lt;small&gt;($escapetool.xml($userName))
+&lt;/small&gt;
+#end
+#end
+#set($headline = $headline.replaceFirst('__headlinesmall__',$!headlineSmall))
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml]:
```suggestion
#set($headline = $headline.replaceFirst('__headlinesmall__', $!headlineSmall))
```

Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml]: `replaceFirst` is regex based which is not useful here. https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#replaceFirst-java.lang.String-java.lang.String-

WDYT of the version below?
```suggestion
#set($headline = $stringtool.replaceOnce($headline, '__headlinesmall__', $headlineSmall))
```

File: xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml [hunk 3]
```diff
@@ -528,19 +528,22 @@
...
 
 {{html}}
 &lt;div class="registration-success-headline"&gt;
-  &lt;h2&gt;$escapetool.xml($headline) &lt;/h2&gt;
-  #if(!$noReadableName)
-  &lt;p class="registration-success-subtitle"&gt;
-    ($escapetool.xml($userName))
-  &lt;/p&gt;
-  #end
+  &lt;h2&gt;$headline &lt;/h2&gt;
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml]: Did you check that there is no space before the `!` when `$noReadableName` is false?
A page test would easily cover this corner case ;)

File: xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties [hunk 4]
```diff
@@ -5718,12 +5715,12 @@ xe.userdirectory.canCustomizeInfoGuest=The user directory can be customized to d
 #######################################
 ## until 18.6.0RC1
 #######################################
-## Renamed to core.editors.content.title.text to avoid using HTML markup in translation messages.
-core.editors.content.title=Editing title and content of <a href="{1}">{0}</a>
 admin.pageandchildrenrights.info=These rights apply on this page {0}and all {1}its children{2}{3}.
 admin.pagerights.info=These rights apply on this page only.
 admin.pagerights.infoNonTerminalDoc=They do not affect the {0}children{1}.
 core.edit.autosave.every=every
+#@deprecated core.register.successful.welcome.headline
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties:5734]: Looks good. Just mentioning that we'll need to add a new "until" section if the fix is merged after 18.6.0RC1.

File: xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml (hunk 2 above)
Comment (Sereza5) [reply xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml]: I agree that this solution is better for optimization. Note that we should remember it for next times. AFAICS, we used exclusively replaceFirst for this kind of substitution so far:
https://github.com/xwiki/xwiki-platform/blob/76b653058d9b9fc1c7b2f53f425724a46ed546ea/xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-api/src/main/resources/templates/extension/security/liveData/advice.vm#L56

Comment (Sereza5) [reply xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml]: Addressed in https://github.com/xwiki/xwiki-platform/pull/4014/commits/5ecd22d58213df5eac50de62d87f79fc493657d3 :+1:

File: xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml (hunk 3 above)
Comment (Sereza5) [reply xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml]: There's a space in both cases, tbh I think it looks better with a space. Note that if we want to remove the space, we'll need a trim on the headlineSmall block and a change in the translation...
I'll leave it as is for now, let me know if you think this is wrong.


image

**Comment by Sereza5** [comment]
This PR is ready for further review :)

File: xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml (hunk 3 above)
Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml]: That's not your choice, but the typographic rules that must prime ;)
In French, a (non-breakable) space must be placed before each `!` (see https://www.scribbr.fr/elements-linguistiques/les-espaces/).
The same is not true in English. Ultimately, that's the translator's responsibility to comply with the language's rules.

File: xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties (hunk 4 above)
Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties:5734]: Update: should be in a 18.7.0RC1 now.

File: xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties [hunk 5]
```diff
@@ -1709,7 +1706,7 @@ core.register.invalidUsername=Invalid username provided. Please use only letters
 core.register.mailSenderWronglyConfigured=The user has been created but the validation email has not been sent. Please check the Mail Sending Configuration and consider recreating the user.
 core.register.invalidCaptcha=Incorrect CAPTCHA answer.
 core.register.registerFailed=Registration has failed due to unknown reasons. (Error code: {0})
-core.register.successful.welcome=Welcome {0}
+core.register.successful.welcome.headline=Welcome {0} {1}!
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties]: With the space part of the translation, I think we still end up with a space before the `!` when `{1}` is the empty string.
I think the fix is to make the translation value `Welcome {0}{1}!` (no space between the two arguments) and `Bienvenu {0}{1} !` in french witht he madatory space (technically it must even but a non-breaking space), and to let the velocity code add the space when there is a value for `{1}`.

File: xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml (hunk 3 above)
Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/RegistrationConfig.xml]: I don't think a trailing space is needed here.
```suggestion
  &lt;h2&gt;$headline&lt;/h2&gt;
```

File: xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties (hunk 4 above)
Comment (Sereza5) [reply xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties:5734]: Addresed in https://github.com/xwiki/xwiki-platform/pull/4014/commits/c6c5ef2afd3d45928aeb0262e8fcb20524ec8fff :+1:

File: xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties (hunk 5 above)
Comment (Sereza5) [reply xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties]: Addressed in https://github.com/xwiki/xwiki-platform/pull/4014/commits/814bc07eb16dacd6dd7b15f4eb8ea1d1e23815df :+1:

File: xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties [hunk 6]
```diff
@@ -5728,6 +5728,12 @@ admin.pagerights.info=These rights apply on this page only.
 admin.pagerights.infoNonTerminalDoc=They do not affect the {0}children{1}.
 core.edit.autosave.every=every
 
+#######################################
+## until 18.7.0RC1
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties:5732]: Sadly we are too close to the release of the RC to merge this PR. I suggest moving it to 18.7.0 (and don't forget to add the stable-18.7.x label).
